service/src/test/java/com/epam/rft/atsy/service/impl/CandidateServiceImplTest.java

```diff
@@ -0,0 +1,212 @@
+
+    private CandidateDTO dummyCandidateDto;
+
+    private FilterRequest ascendingFilterRequest;
+
+    private FilterRequest descendingFilterRequest;
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Before
+    public void setUp() {
+        dummyCandidateEntity = CandidateEntity.builder().id(ID).name(NAME).email(EMAIL).phone(PHONE)
+                                              .referer(REFERER).languageSkill(LANGUAGE_SKILL)
+                                              .description(DESCRIPTION).build();
+
+        dummyCandidateDto = CandidateDTO.builder().id(ID).name(NAME).email(EMAIL).phone(PHONE)
+                                        .referer(REFERER).languageSkill(LANGUAGE_SKILL)
+                                        .description(DESCRIPTION).build();
+
+        ascendingFilterRequest = FilterRequest.builder().fieldName(SORT_FIELD).order(SortingRequest.Order.ASC)
+                                              .searchOptions(new SearchOptions(NAME, EMAIL, PHONE)).build();
+
+        descendingFilterRequest = FilterRequest.builder().fieldName(SORT_FIELD).order(SortingRequest.Order.DESC)
+                                              .searchOptions(new SearchOptions(NAME, EMAIL, PHONE)).build();
+    }
+
+    @Test
+    public void getCandidateShouldReturnNullWhenIdNotFound() {
+        // Given
+        given(candidateRepository.findOne(ID)).willReturn(NOT_FOUND_CANDIDATE_ENTITY);
+        given(modelMapper.map(NOT_FOUND_CANDIDATE_ENTITY, CandidateDTO.class)).willReturn(NOT_FOUND_CANDIDATE_DTO);
+
+        // When
+        CandidateDTO candidate = candidateService.getCandidate(ID);
+
+        // Then
+        assertThat(candidate, nullValue());
+    }
+
+    @Test
+    public void getCandidateShouldReturnProperDtoWhenIdExists() {
+        // Given
+        given(candidateRepository.findOne(ID)).willReturn(dummyCandidateEntity);
+        given(modelMapper.map(dummyCandidateEntity, CandidateDTO.class)).willReturn(dummyCandidateDto);
+
+        // When
+        CandidateDTO candidate = candidateService.getCandidate(ID);
+
+        // Then
+        assertThat(candidate, equalTo(dummyCandidateDto));
+    }
+
+    @Test
+    public void getAllCandidateShouldCallRepositoryWithAscendingSortWhenPassingAscendingFilterRequest() {
+        // Given
+        Sort ascendingSort = new Sort(Sort.Direction.ASC, SORT_FIELD);
+
+        // When
+        Collection<CandidateDTO> result = candidateService.getAllCandidate(ascendingFilterRequest);
+
+        // Then
+        ArgumentCaptor<String> nameArgumentCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<String> emailArgumentCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<String> phoneArgumentCaptor = ArgumentCaptor.forClass(String.class);
+        ArgumentCaptor<Sort> sortArgumentCaptor = ArgumentCaptor.forClass(Sort.class);
+
+        verify(candidateRepository).findAllCandidatesByFilterRequest(
```

Comment:
Without argument captor it would look like the single line:

> verify(candidateRepository).findAllCandidatesByFilterRequest(NAME, EMAIL, PHONE, ascendingSort);